src/toil/batchSystems/torque.py

```diff
@@ -0,0 +1,168 @@
+            # down to integer job ID only may be bad long-term
+            result = int(so.strip().split('.')[0])
+            return result
+
+        def getJobExitCode(self, torqueJobID):
+            args = ["qstat", "-f", str(torqueJobID)]
+
+            process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            for line in process.stdout:
+                line = line.strip()
+                if line.startswith("failed") and int(line.split()[1]) == 1:
+                    return 1
+                if line.startswith("exit_status"):
+                    status = line.split(' = ')[1]
+                    logger.debug('Exit Status: ' + status)
+                    return int(status)
+            return None
+
+        """
+        Implementation-specific helper methods
+        """
+        def prepareQsub(self, cpu, mem, jobID):
+
+            # TODO: passing $PWD on command line not working for -d, resorting to
+            # $PBS_O_WORKDIR but maybe should fix this here instead of in script?
+
+            # TODO: we previosuly trashed the stderr/stdout, as in the commented
+            # code, but these may be retained by others, particularly for debugging.
+            # Maybe an option or attribute w/ a location for storing the logs?
+
+            # qsubline = ['qsub', '-V', '-j', 'oe', '-o', '/dev/null',
+            #             '-e', '/dev/null', '-N', 'toil_job_{}'.format(jobID)]
+
+            qsubline = ['qsub', '-V', '-N', 'toil_job_{}'.format(jobID)]
+
+            if self.boss.environment:
+                qsubline.append('-v')
+                qsubline.append(','.join(k + '=' + quote(os.environ[k] if v is None else v)
+                                         for k, v in self.boss.environment.iteritems()))
+
+            reqline = list()
+            if mem is not None:
+                memStr = str(mem / 1024) + 'K'
+                reqline += ['-l mem=' + memStr]
+
+            if cpu is not None and math.ceil(cpu) > 1:
+                qsubline.extend(['-l ncpus=' + str(int(math.ceil(cpu)))])
+
+            return qsubline
+
+        def generateTorqueWrapper(self, command):
+            """
+            A very simple script generator that just wraps the command given; for
+            now this goes to default tempdir
+            """
+            _, tmpFile = tempfile.mkstemp(suffix='.sh', prefix='torque_wrapper')
+            fh = open(tmpFile , 'w')
+            fh.write("$!/bin/bash\n\n")
+            fh.write("cd $PBS_O_WORKDIR\n\n")
+            fh.write(command + "\n")
+            fh.close
+            return tmpFile
+
+    """
+    The interface for the PBS/Torque batch system
+    """
+
+    @classmethod
+    def obtainSystemConstants(cls):
+
+        maxCPU = 0
+        maxMEM = MemoryString("0K")
+
+        # parse XML output from pbsnodes
+        root = ET.fromstring(subprocess.check_output(["pbsnodes","-x"]))
```

Comment:
@cjfields I know, I didn't explain myself right: PBSPro does not have neither XML nor JSON (I would expect that a paid solution would, but oh well...). 

The -x flag is reserved for sth else in PBSPro and your command does not output XML but just plain qstat outputs:

```
-x        Displays status information for finished and moved jobs in addition 
to running and queued jobs.
```

But anyway, I'm trying to fix the `qstat` info gathering function(s) from `torque.py` *without* breaking your output parsing in the process, that's why I enquired back to you. For instance, my `qstat` output looks like this:

```
Job id            Name             User              Time Use S Queue
----------------  ---------------- ----------------  -------- - -----

687694.r-man2     name1      user1                   0 H normal-def
4589526.r-man2    name2      user2            584:51:0 R gpu-def
4595282.r-man2    name3          user3                   0 W copyq
```

When adding debugging loggers in `getRunningJobIDs` I can see that the times are not gathered properly by Toil's leader which according to @cket might be the reason my master is waiting indefinitely since it's not getting timing feedback from the queue system: 

```
getRunningJobIDs current jobs are: {'4780506.r-man2': 0}
Job times from qstat are: {}
No activity, sleeping for 1s
Job times from qstat are: {}
No activity, sleeping for 1s
Job times from qstat are: {}
No activity, sleeping for 1s
(...)
```

So ideally I would like to have both vanilla qstat outputs and make it work for me *and* you ;)